test/mocks/osx/MockPSP.sol

```diff
@@ -0,0 +1,706 @@
+        //     _params.pluginSetupRef.versionTag
+        // );
+
+        permissions = PluginSetup(popSetup()).prepareUninstallation(_dao, _params.setupPayload);
+
+        // bytes32 preparedSetupId = _getPreparedSetupId(
+        //     _params.pluginSetupRef,
+        //     hashPermissions(permissions),
+        //     ZERO_BYTES_HASH,
+        //     bytes(""),
+        //     PreparationType.Uninstallation
+        // );
+
+        // // Check if this setup has already been prepared before and is pending.
+        // if (pluginState.blockNumber < pluginState.preparedSetupIdToBlockNumber[preparedSetupId]) {
+        //     revert SetupAlreadyPrepared({preparedSetupId: preparedSetupId});
+        // }
+
+        // pluginState.preparedSetupIdToBlockNumber[preparedSetupId] = block.number;
+
+        // emit UninstallationPrepared({
+        //     sender: msg.sender,
+        //     dao: _dao,
+        //     preparedSetupId: preparedSetupId,
+        //     pluginSetupRepo: _params.pluginSetupRef.pluginSetupRepo,
+        //     versionTag: _params.pluginSetupRef.versionTag,
+        //     setupPayload: _params.setupPayload,
+        //     permissions: permissions
+        // });
+    }
+
+    /// @notice Applies the permissions of a prepared uninstallation to a DAO.
+    /// @param _dao The address of the DAO.
+    /// @param _dao The address of the uninstalling DAO.
+    /// @param _params The struct containing the parameters for the `applyUninstallation` function.
+    /// @dev The list of `_params.setupPayload.currentHelpers` has to be specified in the same order as they were returned from previous setups preparation steps (the latest `prepareInstallation` or `prepareUpdate` step that has happend) on which the uninstallation was prepared for.
+    function applyUninstallation(address _dao, ApplyUninstallationParams calldata _params)
+        external
+        canApply(_dao, APPLY_UNINSTALLATION_PERMISSION_ID)
+    {
+        // bytes32 pluginInstallationId = _getPluginInstallationId(_dao, _params.plugin);
+
+        // PluginState storage pluginState = states[pluginInstallationId];
+
+        // bytes32 preparedSetupId = _getPreparedSetupId(
+        //     _params.pluginSetupRef,
+        //     hashPermissions(_params.permissions),
+        //     ZERO_BYTES_HASH,
+        //     bytes(""),
+        //     PreparationType.Uninstallation
+        // );
+
+        // validatePreparedSetupId(pluginInstallationId, preparedSetupId);
+
+        // // Since the plugin is uninstalled, only the current block number must be updated.
+        // pluginState.blockNumber = block.number;
+        // pluginState.currentAppliedSetupId = bytes32(0);
+
+        // Process the permissions, which requires the `ROOT_PERMISSION_ID` from the uninstalling DAO.
+        if (_params.permissions.length > 0) {
+            DAO(payable(_dao)).applyMultiTargetPermissions(_params.permissions);
+        }
+
+        // emit UninstallationApplied({
```

Comment:
Also note: we probably should emit these events so these contracts can be tested on UIs

Reply:
I would expect UI's to be using the officially deployed OSx addresses, rather than a mock used for internal testing